### PR TITLE
Make constants exposable in metadata

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Comment by xlc** [comment]
This looks like going to be heavily conflict with #2880....

File: srml/metadata/src/lib.rs [hunk 1]
```diff
@@ -313,8 +307,10 @@ pub enum RuntimeMetadata {
 	V3(RuntimeMetadataDeprecated),
 	/// Version 4 for runtime metadata. No longer used.
 	V4(RuntimeMetadataDeprecated),
-	/// Version 5 for runtime metadata.
-	V5(RuntimeMetadataV5),
+	/// Version 5 for runtime metadata. No longer used.
+	V5(RuntimeMetadataDeprecated),
+	/// Version 6 for runtime metadata. No longer used.
```
Comment (yjkimjunior) [inline srml/metadata/src/lib.rs]:
```suggestion
	/// Version 6 for runtime metadata.
```

**Comment by Kianenigma** [comment]
I would also place an example of this new constant-configs in the node-template with some some minimal related docs.

File: srml/democracy/src/lib.rs [hunk 2]
```diff
@@ -318,6 +317,28 @@ decl_event!(
 
 decl_module! {
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		/// The minimum period of locking and the period between a proposal being approved and enacted.
```
Comment (gavofyork) [inline srml/democracy/src/lib.rs]: looks like we're duplicating every const definition. was this intended?! not cool.

Comment (gavofyork) [reply srml/democracy/src/lib.rs]: ahh; i see. this is instead of the `decl_properties` macro...

**Comment by gavofyork** [comment]
(doesn't build)

**Comment by bkchr** [comment]
> Fine as a stop-gap solution, but definitely one to be clean up when we move to procedural macros.

Even with a procedural macro we could not make it better. The problem is that we don't have specialization... I don't wanted that the user has to specify in `construct_runtime!`, if a module exposes constants.

File: srml/metadata/src/lib.rs [hunk 3]
```diff
@@ -347,12 +342,14 @@ pub struct RuntimeMetadataV5 {
 pub struct ModuleMetadata {
 	pub name: DecodeDifferentStr,
 	pub prefix: DecodeDifferent<FnEncode<&'static str>, StringBuf>,
-	pub storage: ODFnA<StorageFunctionMetadata>,
+	pub storage: ODFnA<StorageEntryMetadata>,
 	pub calls: ODFnA<FunctionMetadata>,
 	pub event: ODFnA<EventMetadata>,
+	pub constants: DFnA<ModuleConstantMetadata>,
```
[the remark this answers is not in the document]
Comment (bkchr) [reply srml/metadata/src/lib.rs]: The problem is that I don't want to have the user to specify in `construct_runtime!` if a module exposes constants.
So, the function is always available and the encoded data is shorter by 1 byte ;)

**Pull request merged**